main.go

```diff
@@ -61,11 +60,25 @@ func main() {
 		},
 	}
 
-	exitStatus, err := c.Run()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Error executing CLI: %s\n", err.Error())
-		os.Exit(1)
-	}
+	sigs := make(chan os.Signal, 1)
+	done := make(chan int, 1)
+
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		<-sigs
+		done <- 0
+	}()
+
+	go func() {
+		exitStatus, err := c.Run()
```

Comment:
Putting this in a goroutine may seem a bit odd.  Initially, I tried keeping this as part of the implicit/main goroutine, only spinning off another goroutine for the signal handler.  In the signal handler goroutine, I was calling plugin.CleanupClients(), then doing an os.Exit().  I ran into rather odd behavior, where the plugin.CleanupClients() would invoke, but the system would exit before plugin.CleanupClients() finished.